### PR TITLE
Refactored ConfigFileFinder.cs 

### PR DESCRIPTION
**Pull request by sirh3e**

### CODE REVIEW

File: src/Microsoft.Tye.Core/ConfigFileFinder.cs [hunk 1]
```diff
@@ -23,12 +23,12 @@ public static bool TryFindSupportedFile(string directoryPath, out string? filePa
...
                 }
 
-                if (files.Length > 1)
-                {
-                    errorMessage = $"More than one matching file was found in directory '{directoryPath}'.";
-                    filePath = default;
-                    return false;
-                }
+                if (files.Length <= 1)
+                    continue;
```
Comment (jkotalik) [inline src/Microsoft.Tye.Core/ConfigFileFinder.cs:27]: nit prefer braces.

**Review by jkotalik — APPROVED** [review]

**Comment by sirh3e** [comment]
Added braces

**Comment by dasiths** [comment]
There is something wrong with the build pipeline. The Build Windows task timed out on this PR as well. (This happened with with #767 too).

File: src/Microsoft.Tye.Core/ConfigFileFinder.cs [hunk 2]
```diff
@@ -23,12 +23,14 @@ public static bool TryFindSupportedFile(string directoryPath, out string? filePa
                     return true;
                 }
 
-                if (files.Length > 1)
+                if (files.Length <= 1)
```
Comment (JunTaoLuo) [inline src/Microsoft.Tye.Core/ConfigFileFinder.cs:26]: Isn't this condition only applicable when the length is 0? If so let's just make it `if (files.Length == 0)`

**Review by JunTaoLuo — APPROVED** [review]
One nit. Otherwise ready to merge.

**Comment by sirh3e** [comment]
@JunTaoLuo Yeah, you're right. It was late when I made the change. 😄

File: src/Microsoft.Tye.Core/ConfigFileFinder.cs [hunk 3]
```diff
@@ -8,27 +8,27 @@ namespace Microsoft.Tye
...
+        private static readonly string[] FileFormats = { "tye.yaml", "tye.yml", "*.csproj", "*.fsproj", "*.sln" };
 
         public static bool TryFindSupportedFile(string directoryPath, out string? filePath, out string? errorMessage)
         {
             foreach (var format in FileFormats)
             {
                 var files = Directory.GetFiles(directoryPath, format);
 
-                if (files.Length == 1)
+                switch (files.Length)
```
Comment (jkotalik) [inline src/Microsoft.Tye.Core/ConfigFileFinder.cs:19]: nit: move >1 case into default for switch statement.

**Pull request merged**